LanguageFeatures/Patterns/type_inference_A17_t06.dart

```diff
@@ -39,8 +37,4 @@ main() {
 
   final [...List<double> v5] = [pi];
   Expect.listEquals([3.14], v5);
-
-  List<int Function(int)> list = [foo];
-  var <int Function(int)>[v6] = list;
-  v6.expectStaticType<Exactly<int Function(int)>>();
```

Comment:
This one should also continue to work: `[foo]` is inferred as `<int Function(int)>[foo<int>]`, and there are no further type mismatches.